### DISCUSSION TITLE
UML class relationships

### DISCUSSION THREAD

**Discussion by kbuffardi** [terrastruct/d2#1858] [Feature Requests]
I'm starting to use d2 more and more for class diagrams, but I'm running into roadblocks when using some different relationships between classes.

UML supports the following:
![](https://upload.wikimedia.org/wikipedia/commons/thumb/9/93/Uml_classes_en.svg/600px-Uml_classes_en.svg.png)

I've found I can do:
  * Association: 
    * `ClassA -- ClassB`
  * Inheritance (technically, it should be an empty arrow, but d2 creates a filled arrow which is "close enough" by my book): 
    * `ChildClass -> ParentClass`

However, I couldn't figure out if/how the others are supported. It'd be great if the following worked:

```
ClassA - - ClassB  /* Dependency */
ClassA - -> ClassB /* Realization */
ClassA -<> ClassB /* Aggregation */
ClassA -<-> ClassB /* Composition */
```

This was asked [about a year ago but there weren't any responses](https://github.com/terrastruct/d2/discussions/1349)

**Comment by alixander** [comment]
https://d2lang.com/tour/connections/#arrowheads
